Sources/Sentry/SentrySubClassFinder.m

```diff
@@ -24,115 +25,73 @@ - (instancetype)initWithDispatchQueue:(SentryDispatchQueueWrapper *)dispatchQueu
     return self;
 }
 
-- (void)actOnSubclassesOf:(Class)parentClass block:(void (^)(Class))block
+- (void)actOnSubclassesOfViewControllerInImage:(NSString *)imageName block:(void (^)(Class))block;
 {
     [self.dispatchQueue dispatchAsyncWithBlock:^{
-        Class *classes = NULL;
-        int numClasses = -1;
-        int attemptsForGettingClasses = 2;
-
-        // The number of classes may change between the two invocations of class_getSuperclass. If
-        // this or any other error happens, we retry once. We don't want to retry this in a loop
-        // because of the danger of an endless loop.
-        for (int i = 0; numClasses == -1 && i < attemptsForGettingClasses; i++) {
-            numClasses = [self getClassList:&classes];
-
-            if (numClasses < 0) {
-                free(classes);
-                classes = NULL;
-            }
-        }
-
-        if (numClasses < 0) {
-            NSString *msg =
-                [NSString stringWithFormat:@"Not able to get subclasses for %@", parentClass];
-            [SentryLog logWithMessage:msg andLevel:kSentryLevelError];
+        Class viewControllerClass = NSClassFromString(@"UIViewController");
+        if (viewControllerClass == nil) {
+            [SentryLog logWithMessage:@"UIViewController class not found."
+                             andLevel:kSentryLevelDebug];
             return;
         }
 
+        unsigned int count = 0;
+        const char **classes = [self.objcRuntimeWrapper
+            copyClassNamesForImage:[imageName cStringUsingEncoding:NSUTF8StringEncoding]
+                            amount:&count];
+#if TEST
         // Only for testing. We want to know in tests if the code iterated over the classes, because
         // iterating in edge cases could lead to crashses. Ideally, we would wrap
         // class_getSuperclass in the SentryObjCRuntimeWrapper and count its invocations. As
         // class_getSuperclass is called in a tight loop doing so would slow down the code
         // significantly. This is pragmatic workaround to find out in tests if the code iterated
         // over the classes.
         [self.objcRuntimeWrapper countIterateClasses];
```

Comment:
`m`: I think we can remove this and also the implementation of `countIterateClasses` now because it's not needed anymore with the current implementation. Previously, I needed a way to validate how often we iterated over the classes.